hummingbot/connector/exchange/huobi/huobi_exchange.py

```diff
@@ -0,0 +1,423 @@
+                )
+                await self._order_tracker.process_order_not_found(tracked_order.client_order_id)
+                continue
+            if order_update is None:
+                continue
+            if order_update.get("status") != "ok":
+                self.logger().network(
+                    f"Error fetching status update for the order {tracked_order.client_order_id}: {order_update}.",
+                    app_warning_msg=f"Failed to fetch status update for the order {tracked_order.client_order_id}."
+                )
+                continue
+            order_state = order_update["data"]["state"]
+            # possible order states are "submitted", "partial-filled", "filled", "canceled"
+            if order_state not in ["submitted", "filled", "canceled", "partial-filled", "created", "partial-canceled", "canceling"]:
+                self.logger().debug(f"Unrecognized order update response - {order_update}")
+
+            # Calculate the newly executed amount for this update.
+            new_state = CONSTANTS.ORDER_STATE[order_state]
+            update = OrderUpdate(
+                client_order_id=tracked_order.client_order_id,
+                exchange_order_id=str(tracked_order.exchange_order_id),
+                trading_pair=tracked_order.trading_pair,
+                update_timestamp=self.current_timestamp * 1e-3,
+                new_state=new_state,
+            )
+            self._order_tracker.process_order_update(update)
+
+    async def _iter_user_event_queue(self) -> AsyncIterable[Dict[str, Any]]:
+        """
+        Called by _user_stream_event_listener.
+        """
+        while True:
+            try:
+                yield await self._user_stream_tracker.user_stream.get()
+            except asyncio.CancelledError:
+                raise
+            except Exception as e:
+                self.logger().error(f"Unknown error. Retrying after 1 second. {e}", exc_info=True)
+                await asyncio.sleep(1.0)
+
+    async def _user_stream_event_listener(self):
+        """
+        This functions runs in background continuously processing the events received from the exchange by the user
+        stream data source. It keeps reading events from the queue until the task is interrupted.
+        The events received are balance updates, order updates and trade events.
+        """
+        async for stream_message in self._iter_user_event_queue():
+            try:
+                channel = stream_message["ch"]
+                if not channel.startswith("accounts") and not channel.startswith("orders") and not channel.startswith("trade.clearing"):
+                    continue
+                data = stream_message["data"]
+                if len(data) == 0 and stream_message["code"] == 200:
+                    # This is a subcribtion confirmation.
+                    self.logger().info(f"Successfully subscribed to {channel}")
+                    continue
```

Comment:
The subscription confirmations should be filtered out in the UserStreamDataSource class, not here.